runelite-client/src/main/java/net/runelite/client/ui/overlay/components/MinimapOrb.java

```diff
@@ -0,0 +1,261 @@
+/*
+ * Copyright (c) 2018 Abex
+ * Copyright (c) 2016-2017, Adam <Adam@sigterm.info>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.ui.overlay.components;
+
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.Font;
+import java.awt.FontMetrics;
+import java.awt.Graphics2D;
+import java.awt.Point;
+import java.awt.RadialGradientPaint;
+import java.awt.Shape;
+import java.awt.geom.Point2D;
+import java.awt.image.BufferedImage;
+import java.io.IOException;
+import javax.imageio.ImageIO;
+import lombok.Getter;
+import lombok.Setter;
+import net.runelite.client.ui.FontManager;
+import net.runelite.client.ui.overlay.RenderableEntity;
+
+public class MinimapOrb implements RenderableEntity
+{
+	private static final int ORB_X = 26;
+	private static final int ORB_Y = 2;
+	private static final int ORB_W = 29;
+
+	private static final Color BACKGROUND_COLOR = new Color(20, 20, 20);
+	private static final Color HIGHLIGHT_COLOR = new Color(255, 255, 255, 50);
+
+	// Frame
+	private static final BufferedImage FRAME;
+
+	static
+	{
+		try
+		{
+			FRAME = ImageIO.read(MinimapOrb.class.getResourceAsStream("minimap_orb_background.png"));
+		}
+		catch (IOException e)
+		{
+			throw new RuntimeException(e);
```

Comment:
Since this is a guice-created class it will just cause the plugin to not be loaded, which seems to be the correct behavior

Reply:
But the cause of this fail will be class not found (or something like that, now I am not sure which one), and that will not provide any information about why it happened.

Reply:
It will just show a IOException in a RuntimeException in a ExceptionInInitializerError (probably in a CreationException in a PluginInstantiationException). If this can't be found something is fubar and whatever plugin needs this class should not be loading. I still feel this is correct behavior.

Reply:
I think the runtime exception here is correct because it indicates a bug. We can add a test perhaps (maybe the existing guice test already would catch this?)

Reply:
I think the guice test will catch this